worksheets/kotlin/1-krumble/index.md

```diff
@@ -0,0 +1,187 @@
+
+*Take some time to look at the different buttons with your coach - where are the different files located in the project? How do you run your app in an emulator or on your device?*
+
+## 2 Layout
+
+If you feel confident with layouts already, or would prefer to focus on purely kotlin this time. Copy paste the code in this file <https://gist.github.com/daniellevass/aa6d4042b950aaf6c0f2345530e4baa4> into your layout. This will give you all the layout elements we'll use later!
+
+Next, we want to open our layout file. On the left find the res folder and open layout. You should be able to double click on the activity_main.xml and open it.
+
+Layouts open in 'design' mode. This gives you some tools to make layouts with a graphical interface, but it's more common to write layouts by hand using the text editor. If you're familiar with Dreamweaver, it's a similar “What you see is what you get” (WYSIWYG) vs code situation.
+
+We're going to be dealing with the XML directly in this tutorial, so find the 'text' tab in the bottom left corner of the design window.
+
+Now you should see some code that looks familiar-ish! Android layouts using XML, which is very very similar to HTML. You should see already we have a `<RelativeLayout>` and a `<TextView>` tag, with some attributes already there. The thing that looks weird is that all the attributes currently are prefixed with the word android or app.
+
+At the moment Android Studio creates a constraint layout by default. These are bit more complicated to start with. So let's delete everything that's there and put this instead:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+</RelativeLayout>
+```
+
+One of the most important features of our cookie clicker will be our cookie - we can create an `<ImageView>`, you will need to provide a height and a width. We also need to provide it an id so we can connect to it later.
+
+```xml
+<ImageView
+    android:id="@+id/imgCookie"
+    android:layout_width="200dp"
+    android:layout_height="200dp"/>
+```
+> id’s need to start with @+id/ - the plus symbol means it assigns the variable name to the current ImageView
+
+In Android we don't use pixels, but instead use `dp` which is a Density-independent Pixel. 1 point might be 1 pixel on a really low resolution device, but might be 4 pixels, or even 8 pixels on a newer device. Take a look at this handy guide Android wrote for more information <https://developer.android.com/guide/practices/screens_support.html>, you can also check out this stack overflow post <https://stackoverflow.com/a/2025541>
+
+
+Because we have a RelativeLayout parent this means we can do some special things to position our cookie on our screen. We can provide an attribute like android:layout_centerHorizontal="true".
```

Comment:
The code should be in backticks (`) to make it easier to spot that it is code to copy and use.